### ISSUE TITLE
Feature Dimension Mismatch 

### ISSUE THREAD

**Issue by fbiying87** [YuchenLiu98/ECCV2022-SDA-FAS#2]
Thanks a lot for sharing your code and model weights. I really appreciate it. However, I got few questions, while testing your test protocols. 

feature_trans = feature_transformer(input)
 _, feature = model(feature_trans, config.norm_flag)

I would like to ask, if the output dimension of the feature_trans is (1,384). How does this fit to the model with the input weights? The first conv_layer already needs a 4-Dim input (bs, 384, x, x).

Could you please give me some hints, as how to use this implementation?

Best Regards,

**Comment by YuchenLiu98** [comment]
Thanks for your question. Instead of using the class token (bs,1,384) as the output of the vision transformer, we output and reshape the visual tokens (bs,256,384) as (bs,384,16,16) for later convolution. The following two functions should be modified in timm.

    def forward_features(self, x):
        B = x.shape[0]
        x = self.patch_embed(x)
        cls_tokens = self.cls_token.expand(B, -1, -1)  
        x = torch.cat((cls_tokens, x), dim=1)
        x = x + self.pos_embed
        x = self.pos_drop(x)
        for blk in self.blocks:
            x = blk(x)
        x = self.norm(x)
        return x[:, 1:]

    def forward(self, x):
        x1 = self.forward_features(x)
        B, new_HW, C = x1.shape
        x1 = x1.transpose(1 ,2).reshape(B, C, int(np.sqrt(new_HW)), int(np.sqrt(new_HW)))  #B C H W
        return x1

**Comment by fbiying87** [comment]
It worked now. Thank you for the clarification! 👍 :-)

Best,
